File: autumn/human-readable-number/go/log.go
package main

import (
   "fmt"
   "math"
)

func NumberFormat(n float64) string {
   n2 := int(math.Log(n) / math.Log(1024))
   n /= math.Pow(1024, float64(n2))
   return fmt.Sprintf("%.3f", n) + []string{"", " k", " M", " G"}[n2]
}

func main() {
   s := NumberFormat(1264)
   println(s)
}

File: autumn/human-readable-number/dart/log.dart
import 'dart:math';

String numberFormat(double d) {
   var e = log(d) / ln10 ~/ 3;
   var f = d / pow(1000, e);
   return f.toStringAsFixed(3) + ['', ' k', ' M', ' G'][e];
}

void main() {
   var s = numberFormat(9012345678);
   print(s == '9.012 G');
}
